src/replica/ConfigParserMySQL.cc

```diff
@@ -186,11 +189,10 @@ void ConfigParserMySQL::_parseDatabases() {
         DatabaseInfo& info = databaseItr.second;
         for (auto&& table : info.tables()) {
             list<SqlColDef>& columns = info.columns[table];
-            _conn->execute("SELECT " + _conn->sqlId("col_name") + "," + _conn->sqlId("col_type") + "  FROM " +
-                           _conn->sqlId("config_database_table_schema") + "  WHERE " +
-                           _conn->sqlEqual("database", database) + "    AND " +
-                           _conn->sqlEqual("table", table) + "  ORDER BY " + _conn->sqlId("col_position") +
-                           " ASC");
+            string const query = _g.select("col_name", "col_type") + _g.from("config_database_table_schema") +
+                                 _g.where(_g.eq("database", database), _g.eq("table", table)) +
+                                 _g.orderBy(make_pair("col_position", "ASC"));
+            _conn->execute(query);
```

Comment:
I think this is still a bit hard to read, maybe something like this would be easier:

    string query =
        "SELECT col_name, col_type FROM config_database_table_schema "
        "WHERE database = ? AND table = ? ORDER BY col_position";
    _conn->execute(query, database, table); // will substitute ?s with database and table name

Reply:
Prepared statements aren't yet supported by the API. I will consider this idea for some later improvement.
Though, I should also note that the way you've proposed it wouldn't work because the identifiers and values need to be quoted differently, and reserved SQL keywords and functions should not be quoted at all. The proposed mechanism doesn't solve this. That was one of the reasons behind this API.

So, the more realistic code (provided the support for the prepared statement was added) could look like this:
```
QueryGenerator g(connector);
string query =
    "SELECT ?, ?
        FROM ?
        WHERE ? = ? AND table = ?
        ORDER BY ?";
_conn->execute(query,
                             g.id("col_name"),  g.id("col_type"),  // SELECT ...                        
                             g.id("config_database_table_schema"),  // FROM ...
                             g.id("database"), database. g.id("table"), table, /// WHERE ...
                             g.id("col_position"); // ORDER BY ...
```
This would result in generating the correct query:
```
SELECT `col_name`, `col_type` FROM `config_database_table_schema`
  WHERE `database` = '<database-name>' AND `table` = '<table-name>'
  ORDER BY `col_position`
```

Reply:
I should also mention that I don't really like the prepared statements (in general, besides specific use cases where the performance of parsing queries during bulk `INSERT` matters) since they don't allow generating queries independently from the code that's executing those. The parameters binding at a point of query execution is a problem here.